src/primitives/transaction.cpp

```diff
@@ -355,6 +353,10 @@ CAmount CTransaction::GetShieldedValueIn() const
         }
     }
```

Comment:
This iterator loop could be changed to a collection loop like the others above. Non-blocking.